mantidproject/mantid
# Only process or load rows if not empty

**Description of work.**
Adds an additional check before attempting to load or process rows so that only non-empty table rows are included.

**Report to:** nobody

**To test:**
[PRdata_no_runs.zip](https://github.com/mantidproject/mantid/files/2812162/PRdata_no_runs.zip)
1. Interfaces -> SANS -> SANS v2
2. Load the sans2d user and batch file in the attached data
3. Click the "insert row" button to add any number of empty rows
4. Click load, process all, or process selected (highlight any rows to be included in processing for this one)
5. No errors or warnings should occur

Fixes #24561 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
